src/Microsoft.ML.Data/Evaluators/RankerEvaluator.cs

```diff
@@ -509,6 +543,48 @@ public void GetSlotNames(ref VBuffer<ReadOnlyMemory<char>> slotNames)
                 slotNames = new VBuffer<ReadOnlyMemory<char>>(UnweightedCounters.TruncationLevel, values);
             }
         }
+
+        public sealed class Result
+        {
+            /// <summary>
+            /// Normalized Discounted Cumulative Gain
+            /// <a href="https://github.com/dotnet/machinelearning/tree/master/docs/images/ndcg.png"></a>
+            /// </summary>
+            public double[] Ndcg { get; }
+
+            /// <summary>
+            /// <a href="https://en.wikipedia.org/wiki/Discounted_cumulative_gain">Discounted Cumulative gain</a>
+            /// is the sum of the gains, for all the instances i, normalized by the natural logarithm of the instance + 1.
+            /// Note that unline the Wikipedia article, ML.Net uses the natural logarithm.
+            /// <a href="https://github.com/dotnet/machinelearning/tree/master/docs/images/dcg.png"></a>
+            /// </summary>
+            public double[] Dcg { get; }
+
+            /// <summary>
+            /// MaxDcgs is the value of <see cref="Dcg"/> when the documents are ordered in the ideal order from most relevant to least relevant.
+            /// In case there are ties in scores, metrics are computed in a pessimistic fashion. In other words, if two or more results get the same score,
+            /// for the purpose of computing DCG and NDCG they are ordered from least relevant to most relevant.
+            /// </summary>
+            public double[] MaxDcg { get; }
+
+            private static T Fetch<T>(IExceptionContext ectx, IRow row, string name)
+            {
+                if (!row.Schema.TryGetColumnIndex(name, out int col))
+                    throw ectx.Except($"Could not find column '{name}'");
+                T val = default;
+                row.GetGetter<T>(col)(ref val);
+                return val;
+            }
+
+            internal Result(IExceptionContext ectx, IRow overallResult)
+            {
+                VBuffer<double> Fetch(string name) => Fetch<VBuffer<double>>(ectx, overallResult, name);
+
+                Dcg = Fetch(RankerEvaluator.Dcg).Values;
+                Ndcg = Fetch(RankerEvaluator.Ndcg).Values;
+                // MaxDcg = Fetch(RankerEvaluator.MaxDcg).Values;
```

Comment:
>// MaxDcg = Fetch(RankerEvaluator.MaxDcg).Values [](start = 16, length = 48)

Uncomment, and retrieve correctly.

Reply:
Removed from metrics, since it is a dataset characteristic. 

---
In reply to: [220800890](https://github.com/dotnet/machinelearning/pull/1068#discussion_r220800890) [](ancestors = 220800890)